flaskr_tests.py

```diff
@@ -62,14 +81,58 @@ def test_edit(self):
             follow_redirects=True)
         self.app.get('/entry=1', follow_redirects=True)
         rv = self.app.post(
-            '/entry=1', 
-            data = dict(title='Bye',
-                        text='Nothing is allowed here'),
+            '/entry=1',
+            data=dict(title='Bye',
+                      text='Nothing is allowed here'),
             follow_redirects=True)
         assert b'HTML' not in rv.data
         assert b'Entry was successfuly edited' in rv.data
         assert b'Bye' in rv.data
 
+    def test_delete(self):
+        """Test entry delete."""
+        self.login('admin', 'default')
+        self.app.post(
+            'add',
+            data=dict(title='Hey',
+                      text='World'),
+            follow_redirects=True)
+        rv = self.app.post('delete=1', follow_redirects=True)
+        assert b'Hey' not in rv.data
+        assert b'The entry was deleted' in rv.data
+
+    def test_move(self):
+        """Test entry move."""
+        self.login('admin', 'default')
+        # post the first entry
+        self.app.post(
+            'add',
+            data=dict(title='First one',
+                      text='text1'),
+            follow_redirects=True)
+        # post the second entry
+        self.app.post(
+            'add',
+            data=dict(title='Second one',
+                      text='text2'),
+            follow_redirects=True)
+        # trying  to  move the last (by sort order) entry down
+        rv = self.app.post('entry=1&move=down', follow_redirects=True)
+        assert b'Unable to' in rv.data
+        assert b'Successfuly' not in rv.data
+        # moving the first (by sort order) entry down
+        rv = self.app.post('entry=2&move=down', follow_redirects=True)
+        assert b'Unable to' not in rv.data
+        assert b'successfuly' in rv.data
+        # testing if the entries have been moved
+        soup = BeautifulSoup(rv.data, 'html.parser')
```

Comment:
That's where API-like server would make testing a lot easier, as you would only have to deal with json responses. BeautifulSoup definitely looks like a hack, but probably something like this is necessary for the integration test.